onedal/linear_model/linear_model.cpp

```diff
@@ -276,23 +276,52 @@ void init_train_hyperparameters(py::module_& m) {
     using namespace dal::linear_regression::detail;
     using train_hyperparams_t = train_parameters<Task>;
 
-    auto cls = py::class_<train_hyperparams_t>(m, "train_hyperparameters")
-                   .def(py::init())
-                   .def("set_cpu_macro_block",
-                        [](train_hyperparams_t& self, int64_t cpu_macro_block) {
-                            self.set_cpu_macro_block(cpu_macro_block);
-                        })
-                   .def("set_gpu_macro_block",
-                        [](train_hyperparams_t& self, int64_t gpu_macro_block) {
-                            self.set_gpu_macro_block(gpu_macro_block);
-                        })
-                   .def("get_cpu_macro_block",
-                        [](const train_hyperparams_t& self) {
-                            return self.get_cpu_macro_block();
-                        })
-                   .def("get_gpu_macro_block", [](const train_hyperparams_t& self) {
-                       return self.get_gpu_macro_block();
-                   });
+    auto cls =
+        py::class_<train_hyperparams_t>(m, "train_hyperparameters")
+            .def(py::init())
+            .def("set_cpu_macro_block",
+                 [](train_hyperparams_t& self, int64_t cpu_macro_block) {
+                     self.set_cpu_macro_block(cpu_macro_block);
+                 })

```

Comment:
I know that you are just following convention of what was done before (so this is not a hard request that must be done, just a possible nice-to-have), and I haven't looked at the hyperparameter implementation before in sklearnex, but shouldn't the equivalent work and simplify the interface? Let me know what you think.

```suggestion
            .def("set_cpu_macro_block", &train_hyperparams_t::set_cpu_macro_block)
```

It worked for me when I did something equivalent here: https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/onedal/common/sycl.cpp#L84

Reply:
Looks like it works too. Modified all the setters to use that syntax.